src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1000 @@
+    !> contains the coordinate lat, lon.
+    !
+    !-----------------------------------------------------------------------
+    function mpas_geotile_mgr_get_tile(mgr, lat, lon, tile) result(ierr)
+        implicit none
+
+        ! Input variables
+        class (mpas_geotile_mgr_type) :: mgr
+        real (kind=RKIND), value :: lat
+        real (kind=RKIND), value :: lon
+        type (mpas_geotile_type), pointer :: tile
+
+        ! Return variable
+        integer :: ierr
+
+        ierr = 0
+        tile => null()
+
+        ! Normalize longitude to be between -pi and pi
+        call normalize_lon(lon)
+
+        if (.not. mgr % search_tile(lat, lon, tile)) then
+            ierr = mgr % add_tile(lat, lon, tile)
+        endif
+
+    end function mpas_geotile_mgr_get_tile
+
+
+    !***********************************************************************
+    !
+    !  private function mpas_geotile_mgr_search_tile => search_tile
+    !
+    !> \brief   Search to see if a tile has already been loaded
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Private function that searches to see if the datatile that contains
+    !> the coordinate lat, lon has already been loaded. If the datatile has been
+    !> loaded, .true. will be returned and tile will point to the mpas_geotile_type 
+    !> that contains the datatile. If the datatile has not been loaded, .false. 
+    !> will be returned and tile will be unassociated.
+    !
+    !-----------------------------------------------------------------------
+    function mpas_geotile_mgr_search_tile(mgr, lat, lon, tile) result(loaded)
+
+        implicit none
+
+        ! Input variables
+        class (mpas_geotile_mgr_type) :: mgr
+        real (kind=RKIND), value :: lat
+        real (kind=RKIND), value :: lon
+        type (mpas_geotile_type), pointer :: tile
+
+        ! Return variable
+        logical :: loaded
+
+        ! Local variables
+        integer, pointer :: tile_nx
+        integer, pointer :: tile_ny
+        character (len=StrKIND) :: fname
+        integer :: x, y
+        integer :: start_x
+        integer :: start_y
+        integer :: ierr
+
+        loaded = .false.
+        tile => null()
+
+        call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+        call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+
+        !
+        ! Using gen_filename, get the tiles start x and y pixel values of the tile
+        !
+        ierr = mgr % gen_filename(lat, lon, fname, start_x, start_y)
+        if (ierr /= 0) then
+            call mpas_log_write("Error generating filename", messageType=MPAS_LOG_ERR)
+            return
+        endif
+
+        !
+        ! Access the tile in the hash table (-1 here as the hash table is from
+        ! 0:tile_nx, and 0:tile_ny).
+        !
+        x = (start_x - 1) / tile_nx
+        if (x < 0 .or. y > size(mgr % hash, 1)) then
```

Comment:
As pointed out by the gfortran compiler, it seems that `y` is indeed being used before it has been defined here.
```
mpas_geotile_manager.F:373:0:                                                                                                        
                                                                                                                                     
  373 |         if (x < 0 .or. y > size(mgr % hash, 1)) then                                                                         
      |                                                                                                                              
Warning: ‘y’ may be used uninitialized in this function [-Wmaybe-uninitialized]
```